### PR TITLE
Update gantt chart UI to display queued state of tasks

### PR DESCRIPTION
**Pull request by yxiao1996**
related: [#28341](https://github.com/apache/airflow/issues/28341)

### Description

This change updates the UI of gantt charts to display queued state of each tasks(represented as gray boxes). Also, tooltips are added to the queued state boxes in gantt charts to provide related info(start time, duration, etc.). Previously, the gantt chart only shows the tasks after they have start. By having queued state tracking in Gantt chart, it would be easier for airflow users to troubleshoot performance issue in DAG's.

### Testing

Tested this change in my local airflow dev environment using example DAG's. 

Screenshot of updated gantt chart of a sample DAG run:

image

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by yxiao1996** [comment]
Looks like static checks failed, will fix.

**Comment by yxiao1996** [comment]
```
Unknown warnings generated:

/usr/local/lib/python3.7/site-packages/airflow/models/base.py:49:[31mDeprecated API features detected! These feature(s) are not compatible with SQLAlchemy 2.0. [32mTo prevent incompatible upgrades prior to updating applications, ensure requirements files are pinned to "sqlalchemy<2.0". [36mSet environment variable SQLALCHEMY_WARN_20=1 to show all deprecation warnings.  Set environment variable 
SQLALCHEMY_SILENCE_UBER_WARNING=1 to silence this message.[0m (Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)

ERROR! There were 1 warnings generated during the import
```

Seems unrelated to my change.

File: airflow/www/static/js/gantt.js [hunk 1]
```diff
@@ -157,9 +164,19 @@ d3.gantt = () => {
         if (!(a.end_date instanceof moment)) {
           a.end_date = moment(a.end_date);
         }
+        if (a.queued_dttm) {
```
Comment (bbovenzi) [inline airflow/www/static/js/gantt.js]:
```suggestion
        if (a.queued_dttm && !(a.queued_dttm instanceof moment)) {
```
You can combine the if statements

File: airflow/www/static/js/gantt.js [hunk 2]
```diff
@@ -212,12 +229,23 @@ d3.gantt = () => {
...
         });
       })
-      .attr('class', (d) => d.state || 'null')
+      .attr('class', (d) => `${d.state || 'null'} all-tasks`)
       .attr('y', 0)
       .attr('transform', rectTransform)
       .attr('height', () => y.rangeBand())
       .attr('width', (d) => d3.max([x(d.end_date.valueOf()) - x(d.start_date.valueOf()), 1]));
 
+    svg.selectAll('.chart')
```
Comment (bbovenzi) [inline airflow/www/static/js/gantt.js:238]: Could we add a comment above this svg and the other one to explain that its painting the instance run and the instance queued?

File: airflow/www/static/js/gantt.js (hunk 1 above)
Comment (yxiao1996) [reply airflow/www/static/js/gantt.js]: will do

File: airflow/www/static/js/gantt.js (hunk 2 above)
Comment (yxiao1996) [reply airflow/www/static/js/gantt.js:238]: sounds good, will add comment here to mark the line painting all instances vs. instance queued states. also will add similar comments in redraw method.

Comment (bbovenzi) [reply airflow/www/static/js/gantt.js:238]: Just need to rebase too, to fix the CI issue. And I think this is looking good.

**Comment by yxiao1996** [comment]
can anyone help me approve the workflow?

**Comment by yxiao1996** [comment]
@bbovenzi Hi Brent, could you please help me take another look?

**Review by bbovenzi — APPROVED** [review]

**Pull request merged**